plugins/modules/event.py

```diff
@@ -0,0 +1,296 @@
+extends_documentation_fragment:
+  - sensu.sensu_go.auth
+notes:
+  - Metric events bypass the store and are sent off to the event pipeline and corresponding event
+    handlers. Read more about this at
+    U(https://docs.sensu.io/sensu-go/latest/reference/events/#metric-only-events)
+options:
+  timestamp:
+    description:
+      - UNIX time at which the event occurred.
+    type: int
+  entity:
+    description:
+      - Name of the entity associated with this event. It must exist before event creation.
+    type: str
+    required: true
+  check:
+    description:
+      - Name of the check associated with this event. It must exist before event creation.
+    type: str
+    required: true
+  check_attributes:
+    type: dict
+    description:
+      - Additional check parameters. Find out more at
+        U(https://docs.sensu.io/sensu-go/latest/reference/events/#check-attributes)
+    suboptions:
+      duration:
+        description: Command execution time in seconds.
+        type: float
+      executed:
+        description: Time that the check request was executed.
+        type: int
+      history:
+        description: Check status history for the last 21 check executions.
+        type: list
+      issued:
+        description: Time that the check request was issued in seconds since the Unix epoch.
+        type: int
+      last_ok:
+        description: The last time that the check returned an OK status (0) in seconds since the Unix epoch.
+        type: int
+      output:
+        description: The output from the execution of the check command.
+        type: str
+      state:
+        description: The state of the check.
+        choices: [ "passing", "failing", "flapping" ]
+        type: str
+      status:
+        description: Exit status code produced by the check.
+        choices: [ "ok", "warning", "critical", "unknown" ]
+        type: str
+      total_state_change:
+        description: The total state change percentage for the check's history.
+        type: int
+  metric_attributes:
+    type: dict
+    description:
+      - Metric attributes. Find out more at
+        U(https://docs.sensu.io/sensu-go/latest/reference/events/#metric-attributes)
+    suboptions:
+      handlers:
+        description:
+          - An array of Sensu handlers to use for events created by the check.
+            Each array item must be a string.
+        type: list
+        elements: str
+      points:
+        description: Metric data points including a name, timestamp, value, and tags.
+        type: list
+        elements: dict
+'''
+
+EXAMPLES = '''
+- name: Create a simple entity
+  entity:
+    auth:
+      url: http://localhost:8080
+    name: awesome_entity
+    entity_class: proxy
+
+- name: Create a simple check
+  check:
+    auth:
+      url: http://localhost:8080
+    name: awesome_check
+    command: echo "Hello world"
+    subscriptions:
+      - checks
+      - also_checks
+    interval: 30
```

Comment:
I think we can skip this whole thing since we only need entity and check names in the current example. Is this a remnant from the good old times when our API was structured in such a way that we required "fully hydrated" objects in event module?

Reply:
I left this as an example so end-user can see that `entity` and `check` must exist, but yeah, that's already mentioned in the docs.